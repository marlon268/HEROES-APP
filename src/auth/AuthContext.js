import { createContext } from "react";

export const AuhContext = createContext();