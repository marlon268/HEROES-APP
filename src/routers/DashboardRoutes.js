import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcScreem } from '../components/dc/DcScreem'
import { HeroeScreen } from '../components/heroes/HeroeScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { NavBar } from '../components/ui/NavBar'

import "../styles/DashboardRoutes.css"

export const DashboardRoutes = () => {
   return (
      <>

         <NavBar />
         <div className="container-all">
            <Switch>
               <Route exact path="/marvel" component={MarvelScreen} />
               <Route exact path="/hero/:heroeId" component={HeroeScreen} />
               <Route exact path="/dc" component={DcScreem} />
               <Route exact path="/search" component={SearchScreen} />

               <Redirect to="/marvel" />
            </Switch>

         </div>
      </>
   )
}
