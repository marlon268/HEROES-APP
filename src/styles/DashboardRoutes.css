.container-all {
   background-color: rgb(250, 250, 250);
   padding: 10px;
   padding-left: 20px;
}

.navbar {
   padding-left: 20px;
   padding-right: 20px;
}
