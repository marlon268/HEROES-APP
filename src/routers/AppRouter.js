import React, { useContext } from 'react'
import {
   BrowserRouter as Router,
   Switch,
} from "react-router-dom";

import { PrivateRoute } from './PrivateRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { AuhContext } from '../auth/AuthContext';
import { PublicRoute } from './PublicRoute';


export const AppRouter = () => {

   const { user: { logged } } = useContext(AuhContext)

   return (
      <Router>
         <div>

            <Switch>
               <PublicRoute
                  path="/login"
                  component={LoginScreen}
                  isAuthenticated={logged}
               />

               <PrivateRoute
                  path="/"
                  component={DashboardRoutes}
                  isAuthenticated={logged}
               />
            </Switch>

         </div>
      </Router>
   )
}
