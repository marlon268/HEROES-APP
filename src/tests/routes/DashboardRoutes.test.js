import React from 'react'
import { mount } from 'enzyme'
import { DashboardRoutes } from "../../routers/DashboardRoutes"
import { AuhContext } from '../../auth/AuthContext'
import { MemoryRouter } from 'react-router'

describe('Pruebas en el componente <DashboardRoutes/>', () => {

   const contextValue = {
      dispatch: jest.fn(),
      user: {
         logged: true,
         name: 'Juanito'
      }
   }



   test('Debe de mostrarse correctamente', () => {

      const wrapper = mount(
         <AuhContext.Provider value={contextValue}>
            <MemoryRouter>
               <DashboardRoutes />
            </MemoryRouter>
         </AuhContext.Provider>
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('.text-info').text().trim()).toBe('Juanito')
   })


})
