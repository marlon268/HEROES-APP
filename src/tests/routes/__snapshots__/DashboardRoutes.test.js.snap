// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <DashboardRoutes/> Debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Juanito
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container-all"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="container1 animate__animated animate__fadeIn"
      >
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-spider"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Spider Man
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Spider Man"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-captain"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Captain America
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Captain America"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-iron"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Iron Man
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Iron Man"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-thor"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Thor
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Thor"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-hulk"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Hulk
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Hulk"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-wolverine"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Wolverine
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Wolverine"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-daredevil"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Daredevil
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Daredevil"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-hawkeye"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Hawkeye
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Hawkeye"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-cyclops"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Cyclops
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Cyclops"
              className="img"
              src=""
            />
          </a>
        </div>
        <div
          className="container-child"
        >
          <a
            href="/hero/marvel-silver"
            onClick={[Function]}
          >
            <h5
              className="absolute superhero"
            >
              Silver Surfer
            </h5>
            <h5
              className="absolute publisher"
            >
              Marvel Comics
            </h5>
            <img
              alt="Silver Surfer"
              className="img"
              src=""
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
